bgp/modules/terms.py

```diff
@@ -113,3 +115,36 @@ def validate_isbn(isbn):
     def __init__(self):
         super(IsbnExtractorModule, self).__init__(self.validate_isbn)
 
+
+class PageTypeProcessor:
+
+    def __init__(self, modules):
+        self.modules = modules
+
+    def run(self, book):
+        utf8_parser = etree.XMLParser(encoding='utf-8')
+        node = etree.fromstring(book.xml.encode('utf-8'), parser=utf8_parser)
+        for x in node.iter('OBJECT'):
+            param = x[0].attrib['value'].split('.')[0] 
+            pageNo = param[-4:]
+            for m in self.modules:
+                self.modules[m].run(x) 
+    @property
+    def results(self):
+        return  
```

Comment:
This `results` is incomplete and should look similar to:

https://github.com/Open-Book-Genome-Project/sequencer/blob/master/bgp/modules/terms.py#L35-L37

Reply:
```suggestion
        return {m: self.modules[m].results for m in self.modules}
```